test/src/unittests/tonal/test_pitchcontours.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env python
+
+# Copyright (C) 2006-2021  Music Technology Group - Universitat Pompeu Fabra
+#
+# This file is part of Essentia
+#
+# Essentia is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the Affero GNU General Public License
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+from numpy import *
+from essentia_test import *
+
+class TestPitchContours(TestCase):
+
+    def testInvalidParam(self):
+        self.assertConfigureFails(PitchContours(), {'binResolution': -1})
+        self.assertConfigureFails(PitchContours(), {'hopSize': -1})
+        self.assertConfigureFails(PitchContours(), {'minDuration': -1})
+        self.assertConfigureFails(PitchContours(), {'peakDistributionThreshold': -1})
+        self.assertConfigureFails(PitchContours(), {'peakFrameThreshold': -1})
+        self.assertConfigureFails(PitchContours(), {'pitchContinuity': -1})
+        self.assertConfigureFails(PitchContours(), {'sampleRate': -1})
+        self.assertConfigureFails(PitchContours(), {'timeContinuity': -1})
+        
+    def testEmpty(self):
+        emptyPeakBins = [[],[]]
+        emptyPeakSaliences = [[],[]]
+        bins, saliences, startTimes, duration = PitchContours()(emptyPeakBins, emptyPeakSaliences)       
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(saliences, [])
+        self.assertEqualVector(startTimes, [])
+        # Previous tests showed small duration of 0.0058 seconds for zero or empty inputs.        
+        self.assertAlmostEqual(duration, 0.0058, 8)
+
+    def testZeros(self):
+        bins, saliences, startTimes, duration = PitchContours()(array(zeros([2,256])), array(zeros([2,256])))      
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(saliences, [])
+        self.assertEqualVector(startTimes, [])
+        # Previous tests showed small duration of 0.0058 seconds for zero or empty inputs.
```

Comment:
Same here, we can precompute the expected duration from the input number of frames.